gamemodes/terrortown/gamemode/client/cl_voice.lua

```diff
@@ -687,12 +686,18 @@ function VOICE.Tick()
 end
 
 ---
--- Returns whether the local @{Player} is speaking
+-- Returns whether a @{Player} is speaking
 -- @note @{Player:IsSpeaking} does not work for local @{Player}
+-- @param Player ply The @{Player} to check, defaults to the local @{Player}
 -- @return boolean
 -- @realm client
-function VOICE.IsSpeaking()
-    return LocalPlayer().speaking
+function VOICE.IsSpeaking(ply)
+    ply = ply or LocalPlayer()
+    if ply == LocalPlayer() then
+        return ply.speaking
+    else
+        return ply:IsSpeaking()
+    end
```

Comment:
```suggestion
    if not ply or ply == LocalPlayer() then
        return LocalPlayer().speaking
    else
        return ply:IsSpeaking()
    end
```

Not sure if my implementation is better, just a suggestion